pkg/crdpuller/discovery.go

```diff
@@ -92,133 +107,161 @@ func (sp *schemaPuller) PullCRDs(context context.Context, resourceNames ...strin
+				protoSchema := sp.models[gvk]
 				if protoSchema == nil {
-					klog.Errorf("no model for %s -- skipping discovery", swaggerSpecDefinitionName)
+					klog.Infof("CRDPuller: Ignoring a resource that has no OpenAPI Schema: %s (%s)", apiResource.Name, gvk.String())
 					continue
 				}
+				swaggerSpecDefinitionName := protoSchema.GetPath().String()
 
-				klog.Infof("processing discovery for resource %s (%s)", apiResource.Name, CRDName)
-
-				schemaProps := apiextensionsv1.JSONSchemaProps{}
-				protoSchema.Accept(&SchemaConverter{
+				var errors []error
+				converter := &SchemaConverter{
 					schemaProps: &schemaProps,
 					schemaName:  swaggerSpecDefinitionName,
-				})
-
-				hasSubResource := func(subResource string) bool {
-					groupResourceNames := apiResourceNames[gv]
-					if groupResourceNames != nil {
-						return groupResourceNames.Has(apiResource.Name + "/" + subResource)
-					}
-					return false
+					visited:     sets.NewString(),
+					errors:      &errors,
 				}
-
-				statusSubResource := &apiextensionsv1.CustomResourceSubresourceStatus{}
-				if !hasSubResource("status") {
-					statusSubResource = nil
+				protoSchema.Accept(converter)
+				if len(*converter.errors) > 0 {
+					klog.Errorf("CRDPuller: Error during the OpenAPI schema import of resource %s (%s) : %v", apiResource.Name, gvk.String(), *converter.errors)
+					continue
 				}
+			}
 
-				scaleSubResource := &apiextensionsv1.CustomResourceSubresourceScale{
-					SpecReplicasPath:   ".spec.replicas",
-					StatusReplicasPath: ".status.replicas",
-				}
-				if !hasSubResource("scale") {
-					scaleSubResource = nil
+			hasSubResource := func(subResource string) bool {
+				groupResourceNames := apiResourceNames[gv]
+				if groupResourceNames != nil {
+					return groupResourceNames.Has(apiResource.Name + "/" + subResource)
 				}
+				return false
+			}
+
+			statusSubResource := &apiextensionsv1.CustomResourceSubresourceStatus{}
+			if !hasSubResource("status") {
+				statusSubResource = nil
+			}
 
-				crd = &apiextensionsv1.CustomResourceDefinition{
-					TypeMeta:   typeMeta,
-					ObjectMeta: objectMeta,
-					Spec: apiextensionsv1.CustomResourceDefinitionSpec{
-						Group: gv.Group,
-						Versions: []apiextensionsv1.CustomResourceDefinitionVersion{
-							{
-								Name: gv.Version,
-								Schema: &apiextensionsv1.CustomResourceValidation{
-									OpenAPIV3Schema: &schemaProps,
-								},
-								Subresources: &apiextensionsv1.CustomResourceSubresources{
-									Status: statusSubResource,
-									Scale:  scaleSubResource,
-								},
-								Served:  true,
-								Storage: true,
+			scaleSubResource := &apiextensionsv1.CustomResourceSubresourceScale{
+				SpecReplicasPath:   ".spec.replicas",
+				StatusReplicasPath: ".status.replicas",
+			}
+			if !hasSubResource("scale") {
+				scaleSubResource = nil
+			}
+
+			crd = &apiextensionsv1.CustomResourceDefinition{
+				TypeMeta:   typeMeta,
+				ObjectMeta: objectMeta,
```

Comment:
Can these just be inlined instead of referencing variables dozens of lines away?

Reply:
Sure. Initially they were used in 2 locations, which explains why they were there.
I'll change it.

Reply:
Fixed